### PR TITLE
Add Feature Mark Task As Done

### PR DESCRIPTION
**Pull request by juliabellmann**
Added functionality till this User-Story. Now I need the merge to revise the complete styling. Afterwards I will add more functions.

### CODE REVIEW

File: components/TaskCard.js [hunk 1]
```diff
@@ -34,57 +37,56 @@ const StyledTaskCard = styled.div`
...
-      };
-    const Trashcan = () => {
-        return (
-            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512">
-            {/* <!--!Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2024 Fonticons, Inc.--> */}
-            <path d="M170.5 51.6L151.5 80l145 0-19-28.4c-1.5-2.2-4-3.6-6.7-3.6l-93.7 0c-2.7 0-5.2 1.3-6.7 3.6zm147-26.6L354.2 80 368 80l48 0 8 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-8 0 0 304c0 44.2-35.8 80-80 80l-224 0c-44.2 0-80-35.8-80-80l0-304-8 0c-13.3 0-24-10.7-24-24S10.7 80 24 80l8 0 48 0 13.8 0 36.7-55.1C140.9 9.4 158.4 0 177.1 0l93.7 0c18.7 0 36.2 9.4 46.6 24.9zM80 128l0 304c0 17.7 14.3 32 32 32l224 0c17.7 0 32-14.3 32-32l0-304L80 128zm80 64l0 208c0 8.8-7.2 16-16 16s-16-7.2-16-16l0-208c0-8.8 7.2-16 16-16s16 7.2 16 16zm80 0l0 208c0 8.8-7.2 16-16 16s-16-7.2-16-16l0-208c0-8.8 7.2-16 16-16s16 7.2 16 16zm80 0l0 208c0 8.8-7.2 16-16 16s-16-7.2-16-16l0-208c0-8.8 7.2-16 16-16s16 7.2 16 16z"/></svg>
-        );
-      };
-
     function getVariant(task) {
```
Comment (JessicaLoers) [inline components/TaskCard.js:90]: - You can simplify  the `getVariant` function
- Then, you can handle the conditional styling directly within `StyledTaskCard` using styled-components.

_Example:_

```js

const StyledTaskCard = styled.div`
// other code 
 
  background-color: ${({ $variant }) =>
    $variant === "overdue"
      ? "var(--bg-overdue)"
      : $variant === "today"
      ? "var(--bg-today)"
      : "var(--bg-default)"};
`

// other code 

//  Determine the variant
  function getTaskVariant(dueDate) {
    const currentDate = new Date().toISOString().split("T")[0];
    const taskDueDate = new Date(dueDate).toISOString().split("T")[0];

    if (currentDate > taskDueDate) return "overdue";
    if (currentDate === taskDueDate) return "today";
    return "default";
  }
  
  // get the Variant
  const taskVariant = getTaskVariant(task.dueDate);
  
  
  return (
    <StyledTaskCard $variant={taskVariant} $isDone={task.isDone}>
  // other code
  )
  

```

**Review by JessicaLoers — COMMENTED** [review]
Great job on the implementation! 🎉  Everything works as expected, and it's clear that you're applying many of the concepts we've discussed.

However, I did notice a few areas where you could refine your approach, particularly with your use of [Styled Components](https://github.com/neuefische/hh-web-24-8/blob/main/sessions/react-styled-components/react-styled-components.md). 

#### Suggestions for improvement

**Combine Styled Components**
- Both `StyledButton` and `StyledDoneButton` share similar properties (e.g., `display`, `align-items`, `justify-content`).
- You can create a generic `BaseButton` styled component that handles shared styles, and extend it for specific button needs.
- If this is too confusing for you, you can also leave it as it is for the time being.  

_Example:_
```js

const BaseButton = styled.button`
    display: flex;
    align-items: center;
    justify-content: center;
    border: 1px solid black;
`;

const StyledDoneButton = styled(BaseButton)`
    padding: 10px;
    position: absolute;
    top: 0;
    right: 2rem;
    height: 30px;
    width: 30px;
    border-top: 0;
    border-radius: 0 0 5px 5px;
`;

```

**Using props to dynamically adjust button styles**
- If you plan to use different buttons with slight variations throughout your project. 

_Example:_
```js
const Button = styled.button`
    display: flex;
    align-items: center;
    justify-content: center;
    padding: ${({ padding }) => padding || "10px"};
    position: ${({ position }) => position || "relative"};
    border: 1px solid black;
    border-radius: ${({ borderRadius }) => borderRadius || "0"};
`;
```

**Highlight Active Links in NavBar**
-Use Next.js `useRouter` to determine the current route and apply styles to the active link for better user navigation experience.

_Example:_
```js
  import { useRouter } from "next/router";

     const StyledNavLink = styled(Link)`
       padding: 10px 30px;
       color: ${({ isActive }) => (isActive ? "blue" : "black")};
       text-decoration: ${({ isActive }) => (isActive ? "underline" : "none")};
     `;

     export default function NavBar() {
       const { pathname } = useRouter();

       return (
         <StyledNav>
           <StyledNavLink href="/" isActive={pathname === "/"}>All Tasks</StyledNavLink>
           <StyledNavLink href="/tasksdone" isActive={pathname === "/tasksdone"}>Done</StyledNavLink>
         </StyledNav>
       );
     }
```

**Issue Head Component in `RootLayout.js`**
The` <Head>` is hardcoded in `RootLayout.js`, which limits its ability to adapt to different pages.

_Current Issue:_
Your` RootLayout.js` has a static `<Head>`:

```js
<Head>
  <title>Workflow Wizzard</title>
</Head>
```

⚠️ _This means:_
- Every page will have the same `<title>`, which is bad for SEO and user experience.

💡_Possible Solution:_
Instead of making `RootLayout.js` responsible for the `<Head>`, you can let individual pages define their `<Head>`.

_Examaple:_
```js
export default function HomePage() {
  return (
    <>
      <Head>
        <title>Home - Workflow Wizzard</title>
        <meta name="description" content="Welcome to the Workflow Wizzard, your task management assistant." />
      </Head>
      <h2>Welcome to your tasks!</h2>
    </>
  );
}
```

You're on the right track. With a little more practice your code will become cleaner and easier to maintain. Keep up the great work—you're doing awesome! 💪

File: components/StyledButton.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    align-items: center;
+
+    background-color: white;
+    border-radius: 1px solid black;
+    min-width: 50px;
+    min-height: 50px;
+
+    display: flex;
+    align-items: center;
+    justify-content: center;
```
[the remark this answers is not in the document]
Comment (juliabellmann) [reply components/StyledButton.js:15]: oh :woman_facepalming: danke

File: components/TaskCard.js (hunk 1 above)
Comment (juliabellmann) [reply components/TaskCard.js:90]: thank you 🌷

**Review by JessicaLoers — APPROVED** [review]
Well done 🎉

**Pull request merged**